tsdb/executor.go

```diff
@@ -622,6 +624,61 @@ func (e *SelectExecutor) close() {
 	}
 }
 
+func (e *SelectExecutor) processTopBottom(results [][]interface{}, columnNames []string) ([][]interface{}, error) {
+	aggregates := e.stmt.FunctionCalls()
+	var call *influxql.Call
+	process := false
+	for _, c := range aggregates {
+		if c.Name == "top" || c.Name == "bottom" {
+			process = true
+			call = c
+			break
+		}
+	}
+	if !process {
+		return results, nil
+	}
+	var values [][]interface{}
+
+	// Check if we have a group by, if not, rewrite the entire result by flattening it out
+	//if len(e.stmt.Dimensions) == 0 {
+	for _, vals := range results {
+		// start at 1 because the first value is always time
+		for j := 1; j < len(vals); j++ {
+			tMin := vals[0].(time.Time)
+			switch v := vals[j].(type) {
+			case influxql.PositionPoints:
+				for _, p := range v {
```

Comment:
Quite a bit of indenting here.  Maybe move to a func and do something like `values = e.toQueryResult(v)`?